stablehlo/dialect/TypeInference.cpp

```diff
@@ -975,8 +980,12 @@ LogicalResult inferSliceOp(Optional<Location> location, Value operand,
     shape.push_back(inferSliceDim(rankedTy.getDimSize(i), start[i], limit[i],
                                   strideVals[i]));
   }
-  inferredReturnTypes.assign(
-      {RankedTensorType::get(shape, rankedTy.getElementType())});
+
+  auto encodingOrErr =
```

Comment:
I think that with boundsToEncoding simplified as discussed above this might already be simple enough: `RankedTensorType::get(shape, elementType, boundsToEncoding(encoding, bounds))`. What do you think?

Reply:
that is neat!

Reply:
Sure!